atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingClosableIterator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl.expectations;
+
+import com.palantir.common.base.ClosableIterator;
+import com.palantir.common.base.ForwardingClosableIterator;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class TrackingClosableIterator<T> extends ForwardingClosableIterator<T> {
+    ClosableIterator<T> delegate;
+    Function<T, Long> measurer;
```

Comment:
measurer is a bit too abstract a name. I cannot suggest one as I am unsure what you were going for.

Reply:
Agreed

I was going for "whatever you can use to measure the size of an object in bytes", thought about measurer/ruler/sizer